gems/aws-sigv4/spec/spec_helper.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+$LOAD_PATH.unshift File.expand_path('../lib', __dir__)
+
+# use the local version of aws-crt libs
+$LOAD_PATH.unshift File.expand_path('../../aws-crt/lib', __dir__)
+$LOAD_PATH.unshift File.expand_path('../../aws-crt-auth/lib', __dir__)
+
+require_relative '../../aws-crt/spec/spec_helper'
+require 'rspec'
+require 'aws-sigv4'
+require 'aws-eventstream'
+
+
+module SpecHelper
+  class << self
+
+    def debug(msg)
+      $stdout.puts("\n#{msg}") if ENV['DEBUG']
+    end
+
+    # @param [String] request
+    # @return [Hash]
+    def parse_request(request)
+      lines = request.lines.to_a
+
+      http_method, request_uri, _ = lines.shift.split
+
+      # escape the uri
+      uri_path, querystring = request_uri.split('?', 2)
+      if querystring
+        querystring = querystring.split('&').map do |key_value|
+          key, value = key_value.split('=')
+          key = Aws::Sigv4::Signer.uri_escape(key)
+          value = Aws::Sigv4::Signer.uri_escape(value.to_s)
+          "#{key}=#{value}"
+        end.join('&')
+      end
+
+      request_uri = Aws::Sigv4::Signer.uri_escape_path(uri_path)
+      request_uri += '?' + querystring if querystring
+
+      # extract headers
+      headers = Hash.new { |h,k| h[k] = [] }
+      prev_key = nil
+      until lines.empty?
+        line = lines.shift
+        if line.strip == ''
+          break
+        elsif line =~ /^\s+/ # multiline header value
+          headers[prev_key] << line.strip
```

Comment:
This test is bad, here's where I fixed it in aws-c-auth

https://github.com/awslabs/aws-c-auth/pull/106/files#diff-4b155fa359124b841f867c72487a57d84358a16d43efea078bfef06c13fb23cf

Reply:
Fixed! Thanks for the reference